### ISSUE TITLE
Polars concat fails to typecheck with ty

### ISSUE THREAD

**Issue by kszlim** [astral-sh/ty#759]
### Summary

Using polars 1.3.1 and ty fails to typecheck the following block with:
```
error[unresolved-attribute]: Type `DataFrame` has no attribute `collect`
 --> r.py:5:7
  |
3 | ldf = pl.LazyFrame({"a": [1,2,3]})
4 | z = pl.concat([ldf, ldf])
5 | print(z.collect())
  |       ^^^^^^^^^
  |
info: rule `unresolved-attribute` is enabled by default
```

MRE:
```python
import polars as pl

ldf = pl.LazyFrame({"a": [1,2,3]})
z = pl.concat([ldf, ldf])
print(z.collect())
```

This could be an issue in polars code for this function is [here](https://github.com/pola-rs/polars/blob/bfcb40e629e4e3a3d2ee4ab8f0653886fa4a7dd8/py-polars/polars/functions/eager.py#L26), but that method does seem to typecheck with pyright. Not sure, but maybe it's related to https://github.com/astral-sh/ty/issues/688.

### Version

0.0.1-alpha.13

**Comment by carljm** [comment]
This is because we currently still infer `list[Unknown]` for the list literal `[ldf, ldf]`. There might also be an issue with how we then resolve the type var in the `concat` call, given an argument of type `list[Unknown]` -- it seems we resolve it to `DataFrame` (I'm not sure why), but we should just consider the return type `Unknown`.

**Comment by AlexWaygood** [comment]
We type-check this correctly now that https://github.com/astral-sh/ruff/pull/20360 has landed

**Comment by kszlim** [comment]
@AlexWaygood is this in the latest pre-release?

**Comment by AlexWaygood** [comment]
> [@AlexWaygood](https://github.com/AlexWaygood) is this in the latest pre-release?

yup!
